tests/runtime/ray/test_build_workflow.py

```diff
@@ -292,22 +292,23 @@ def test_handles_taskwrappingerror(
 
         # Given
         workflow = workflow_parametrised_with_resources().model
+        assert workflow.metadata is not None
         workflow.metadata.python_version = python_version
         client = create_autospec(_build_workflow.RayClient)
         client.add_options = Mock(side_effect=exceptions.TaskWrappingError)
 
         wf_run_id = "<wf run id sentinel>"
 
         # When
-        with pytest.raises(expected_exception):
+        with pytest.raises(expected_exception):  # type: ignore
             _ = _build_workflow.make_ray_dag(client, workflow, wf_run_id, None)
 
     @staticmethod
     def test_pythonversionmismatch_message(monkeypatch: pytest.MonkeyPatch):
         # Given
         workflow = workflow_parametrised_with_resources().model
 
-        workflow.metadata.python_version = ir.Version(
+        workflow.metadata.python_version = ir.Version(  # type: ignore
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏`# type: ignore` should only be used as a last resort.

```suggestion
        assert workflow.metadata is not None
        workflow.metadata.python_version = ir.Version(  # type: ignore
```